### PR TITLE
Support onboard sensors over I2C, SPI buses and Multi Socketcan interface over Rddrone fmuk66 

### PR DESCRIPTION
**Pull request by sumitbatra-nxp**
Added Support for following sensors and multiple socketcan interfaces for rddrone_fmuk66 board-

1. I2C based sensors - bme280 and bmm150
2. SPI based sensors - fxas21002, fxos8700
3. Code to support 2 Parallel SOCKETCAN interfaces.
4. Ensured enabling 3.3V supply for sensors earlier than Sensor Init
5. Add support for servo motors (PWM outputs)

### CODE REVIEW

**Comment by mmahadevan108** [comment]
Does this fix Issue https://github.com/zephyrproject-rtos/zephyr/issues/45218

**Comment by sumitbatra-nxp** [comment]
@mmahadevan108 .. Yes it does resolve [#45218](https://github.com/zephyrproject-rtos/zephyr/issues/45218)

File: boards/arm/rddrone_fmuk66/rddrone_fmuk66.dts [hunk 1]
```diff
@@ -240,15 +289,15 @@ zephyr_udc0: &usbotg {
 	status = "okay";
 	pinctrl-0 = <&flexcan0_default>;
 	pinctrl-names = "default";
-	bus-speed = <125000>;
+	bus-speed = <500000>;
```
Comment (alexanderwachter) [inline boards/arm/rddrone_fmuk66/rddrone_fmuk66.dts:292]: Please revert the bitrate changes. We agreed on 125000 for all boards

**Review by alexanderwachter — CHANGES_REQUESTED** [review]

File: include/zephyr/net/net_if.h [hunk 2]
```diff
@@ -926,6 +926,9 @@ struct net_if *net_if_get_default(void);
  */
 struct net_if *net_if_get_first_by_type(const struct net_l2 *l2);
 
+#if CONFIG_RDDRONE_FMUK66_SOCKETCAN_DEVICES == 2
```
Comment (henrikbrixandersen) [inline include/zephyr/net/net_if.h:929]: We cannot have board specific code in the generic network interface headers.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
- You need to take a look at your Git commit logs and adjust them to match the style used in the rest of the in-tree commits. 
- You need a linear history, no merge commits.
- Please split this PR in multiple PRs, each dealing with a subset. The proposed changes to the network headers and SocketCAN support will not be accepted as-is.

File: include/zephyr/net/net_if.h (hunk 2 above)
Comment (alexanderwachter) [inline include/zephyr/net/net_if.h:929]: This file is generic and should not contain soc specific defined

File: samples/net/sockets/can/src/main.c [hunk 3]
```diff
@@ -23,6 +23,11 @@ LOG_MODULE_REGISTER(net_socket_can_sample, LOG_LEVEL_DBG);
 static k_tid_t tx_tid;
 static K_THREAD_STACK_DEFINE(tx_stack, STACKSIZE);
 static struct k_thread tx_data;
+#if CONFIG_RDDRONE_FMUK66_SOCKETCAN_DEVICES == 2
```
Comment (alexanderwachter) [inline samples/net/sockets/can/src/main.c:26]: Same here. I would prefer some generic device tree based solution.

**Review by alexanderwachter — CHANGES_REQUESTED** [review]

**Comment by landonh12** [comment]
> * You need to take a look at your Git commit logs and adjust them to match the style used in the rest of the in-tree commits.
> * You need a linear history, no merge commits.
> * Please split this PR in multiple PRs, each dealing with a subset. The proposed changes to the network headers and SocketCAN support will not be accepted as-is.

Thanks for the help, we will get this cleaned up.

**Pull request closed without merging**